### ISSUE TITLE
Provide different EfficientHRNet configs details?

### ISSUE THREAD

**Issue by chaowentao** [TeCSAR-UNCC/EfficientHRNet#2]
Hello, I am interested in your work. And I plan to implement your work in [mmpose](https://github.com/open-mmlab/mmpose). I am concerned about different EfficientHRNet configs. Should I change configs from the following figures? If you are convenient, could you share the different EfficientHRNet configs details? Thank you.

> Varying the following parameters provide different EfficientHRNet models ranging from H0 to H-4 :
> 
> scale_factor
> input_size
> width_mult
> depth_mult

![image](https://user-images.githubusercontent.com/37093663/103744180-4fac1380-5038-11eb-97b0-2c79942ad240.png)
![image](https://user-images.githubusercontent.com/37093663/103744215-5d619900-5038-11eb-845a-45459588e057.png)

**Comment by CGNefarious** [comment]
I added a folder with example configurations for our different models so you can see how the config files change with the different models. It follows the table, though if you're creating your own model you should feel free to experiment with different configurations. If you wish to scale up, you can simply change phi in the positive direction, similar to how it's done in EfficientNet.

**Comment by chaowentao** [comment]
@CGNefarious Thank you. I plan to reimplement your work in coco keypoint dataset. If the result is good, I will advise [mmpose](https://github.com/open-mmlab/mmpose) to add your model as a new backbone and let more people know your significant work.

**Comment by chaowentao** [comment]
@CGNefarious Hello, I implement your word base your release code without use EfficientNet pretrained. The result may differ from your paper. By the way, the H-0 model released couldn't be load according to `H-0.yaml`. Can you give me some suggestions?

#### Results on COCO val2017 without multi-scale test
paper
| Arch | Input Size | AP | AP<sup>50</sup> | AP<sup>75</sup> | AR | AR<sup>50</sup> | parms | flops |
| :----------------- | :-----------: | :------: | :------: | :------: | :------: | :------: |:------: |:------: |
| H-0| 512x512 | 0.648 | - | - | - | - | 23.3M | 25.6 |
| H-1| 480x480 | 0.592 | 0.826 | 0.640 | 0.647 | 0.864 | 16.0M | 14.2 |
| H-2| 448x448 | 0.529 | 0.805 | 0.591 | 0.593 | 0.838 | 10.3M | 7.7 |
| H-3| 416x416 | 0.448 | 0.767 | 0.482 | 0.524 | 0.807 | 6.9M | 4.2 |
| H-4| 384x384 | 0.357 | 0.696 | 0.337 | 0.429 | 0.739 | 3.7M | 2.1 |

myself

| Arch | Input Size | AP | AP<sup>50</sup> | AP<sup>75</sup> | AR | AR<sup>50</sup> | parms | flops |
| :----------------- | :-----------: | :------: | :------: | :------: | :------: | :------: |:------: |:------: |
| H-0| 512x512 | 0.573 | 0.800 | 0.624 | 0.623 | 0.824 | 23.3M | 25.6 |
| H-1| 480x480 | 0.531 | 0.770 | 0.571 | 0.582 | 0.800 | 16.0M | 14.2 |
| H-2| 448x448 | 0.463 | 0.736 | 0.489 | 0.521 | 0.768 | 10.3M | 7.7 |
| H-3| 416x416 | 0.348 | 0.665 | 0.326 | 0.412 | 0.702 | 6.9M | 4.2 |
| H-4| 384x384 | 0.209 | 0.535 | 0.115 | 0.272 | 0.589 | 3.7M | 2.1 |

**Comment by CGNefarious** [comment]
The results from our paper all use a backbone EfficientNet model pretrained on ImageNet. Without pretrained weights, the model will likely take longer to converge and you may have to use different training parameters than we present in our paper.

You have brought to my attention that while I provided the weights for EfficientHRNet, I neglected to provide weights for our Compact EfficientNet models trained on ImageNet. I will edit the repository with links to those as well.

As for not being able to load the weights for H0 using the provided configuration file, I assume it was a size mismatch between the weights and the model? Specifically looking for a size of 34 but getting 17? If so, please use the updated H0.yaml in the Example Configs and try again. This is due to the model being trained with AE Loss set to [True, False], so the model must be loaded with the same config parameters. If you still are unable to load the model, let me know the errors you are getting and I will help you figure it out.

**Comment by chaowentao** [comment]
@CGNefarious Thank you for your reply. I will try new pipelines as your mentioned. Best wish to you.

**Comment by chaowentao** [comment]
@CGNefarious Unfortunately,  I can't implement the `H-4.yaml` result. Your `H-4.yaml` use pretrain is: `efficientnet-b0-4cfa50.pth`. I can't load it because size mismatch. I will implement the `H-0.yaml` by load `efficientnet-b0-4cfa50.pth`.

![image](https://user-images.githubusercontent.com/37093663/104540123-d3c65280-5659-11eb-98de-8bc397faf43f.png)

**Comment by CGNefarious** [comment]
That error appears to be due to a mismatch in the tags size. You still get that when using [True, False] in the config file under AE Loss? And do you have the same error loading any of the other models (H-1 through H-4)? If so, I will look into why that's the case. I do not have the issue on my end, but I will try to look into it more by this weekend.

For H-4, you have a size mismatch when loading the BC4 backbone weights? Can you post the error for me to look at?

**Comment by chaowentao** [comment]
@CGNefarious For H-4, I can load BC4 as pretrain model correctly. But the result mAP is 0.197. Also, when I test  `efficientnet-b0-4cfa50.pth`,  the error is  size mismatch as follows: 
![image](https://user-images.githubusercontent.com/37093663/104549839-8ef8e680-566e-11eb-9656-64eb1fc172ef.png)

**Comment by CGNefarious** [comment]
Alright. I made some minor modifications to the code that should fix the error you're getting. Can you pull and try again?

**Comment by chaowentao** [comment]
@CGNefarious Thank you very much. I will try it.

**Comment by xiexiaoshinick** [comment]
According to my experiment，Modify the configuration file **H0.yaml** in the following places to import the model 
`TEST:
  FLIP_TEST: True
  IMAGES_PER_GPU: 1
  MODEL_FILE: 'pretrain_model/H0.pth.tar'  
  SCALE_FACTOR: [1]
  DETECTION_THRESHOLD: 0.1
  WITH_HEATMAPS: (True, True)
  **WITH_AE: (True, True)**
  PROJECT2IMAGE: True
  NMS_KERNEL: 5
  NMS_PADDING: 2
TRAIN:
`

**Comment by chaowentao** [comment]
@xiexiaoshinick Hello, can you reimplement the results from the paper? If you achieve it, please tell me. Thank you.

**Comment by xiexiaoshinick** [comment]
> @xiexiaoshinick Hello, can you reimplement the results from the paper? If you achieve it, please tell me. Thank you.
我尝试在商汤科技的**mmpose**复现EfficientHRNet，训练还没有完成，目前看来，论文的精度结果是可以复现的，训练结束之后我会向mmpose官方申请提交代码😁
